specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-04-01/managedClusters.json

```diff
@@ -1439,6 +1439,10 @@
             "type": "string"
           },
           "description": "(PREVIEW) Availability zones for nodes. Must use VirtualMachineScaleSets AgentPoolType."
+        },
+        "enableNodePublicIP": {
+          "type": "boolean",
```

Comment:
you may evaluate making this value an enum instead of a Boolean, in case you need to support a range of values in the future.